apps/aehttp/src/aehttp_dispatch_ext.erl

```diff
@@ -166,6 +166,248 @@ handle_request('GetGenerationByHeight', Params, _Context) ->
+                {error, _} -> {404, [], #{reason => <<"Contract not found">>}};
+                {ok, Contract} ->
+                    Response = aect_contracts:serialize_for_client(Contract),
+                    {200, [], Response}
+            end
+    end;
+
+handle_request('GetContractCode', Req, _Context) ->
+    case aec_base58c:safe_decode(contract_pubkey, maps:get(pubkey, Req)) of
+        {error, _} -> {400, [], #{reason => <<"Invalid public key">>}};
+        {ok, PubKey} ->
+            case aec_chain:get_contract(PubKey) of
+                {error, _} -> {404, [], #{reason => <<"Contract not found">>}};
+                {ok, Contract} ->
+                    Code = aect_contracts:code(Contract),
+                    {200, [], #{ <<"bytecode">> => aeu_hex:hexstring_encode(Code) }}
+            end
+    end;
+
+handle_request('GetContractStore', Req, _Context) ->
+    case aec_base58c:safe_decode(contract_pubkey, maps:get(pubkey, Req)) of
+        {error, _} -> {400, [], #{reason => <<"Invalid public key">>}};
+        {ok, PubKey} ->
+            case aec_chain:get_contract(PubKey) of
+                {error, _} -> {404, [], #{reason => <<"Contract not found">>}};
+                {ok, Contract} ->
+                    Response = [ #{<<"key">> => aeu_hex:hexstring_encode(K),
+                                   <<"value">> => aeu_hex:hexstring_encode(V)}
+                               || {K, V} <- maps:to_list(aect_contracts:state(Contract)) ],
+                    {200, [], #{ <<"store">> => Response }}
+            end
+    end;
+
+handle_request('GetOracleByPubkey', Params, _Context) ->
+    case aec_base58c:safe_decode(oracle_pubkey, maps:get(pubkey, Params)) of
+        {ok, Pubkey} ->
+            case aec_chain:get_oracle(Pubkey) of
+                {ok, Oracle} ->
+                    {200, [], aehttp_api_parser:encode(oracle, aeo_oracles:serialize_for_client(Oracle))};
+                {error, _} ->
+                    {404, [], #{reason => <<"Oracle not found">>}}
+            end;
+        {error, _} ->
+            {400, [], #{reason => <<"Invalid public key">>}}
+    end;
+
+handle_request('GetOracleQueriesByPubkey', Params, _Context) ->
+    case aec_base58c:safe_decode(oracle_pubkey, maps:get(pubkey, Params)) of
+        {ok, Pubkey} ->
+            Limit = case maps:get(limit, Params) of
+                        N when N =/= undefined -> N;
+                        undefined -> 20
+                    end,
+            FromQueryId = case maps:get(from, Params) of
+                              Id when Id =/= undefined ->
+                                  {ok, OracleQueryId} = aec_base58c:safe_decode(oracle_query_id, Id),
+                                  OracleQueryId;
+                              undefined ->
+                                  '$first'
+                          end,
+            QueryType = case maps:get(type, Params) of
+                            T when T =/= undefined -> T;
```

Comment:
Does [the enum in the Swagger spec](https://github.com/aeternity/epoch/pull/1372/files#diff-89b8ec7c2f28f3e3c7d21ca9bf070bb7R690) validate? Otherwise validation of value in enum should be repeated here (otherwise client mistyping parameter would get 500).

Reply:
All is done on [swagger validation level](https://github.com/aeternity/epoch/blob/PT-158827776-http-api/apps/aehttp/src/aehttp_api_validate.erl#L214) 😉